plugins/modules/open_iscsi.py

```diff
@@ -402,24 +401,32 @@ def main():
         result['nodes'] = nodes
 
     if login is not None:
-        loggedon = target_loggedon(module, target, portal, port)
-        if (login and loggedon) or (not login and not loggedon):
-            result['changed'] |= False
-            if login:
-                result['devicenodes'] = target_device_node(target)
-        elif not check:
-            if login:
-                target_login(module, target, portal, port)
-                # give udev some time
-                time.sleep(1)
-                result['devicenodes'] = target_device_node(target)
+        result['devicenodes'] = []
+        for target in nodes:
```

Comment:
You're overwriting the module paramter `target` here, which was stored in `target` before. You don't seem to use that parameter anymore *at all*.

Reply:
My bad, i think it's just better to make a separate if statement for a not specified target
Thanks!